lib/edge_impulse/ei_wrapper.cpp

```diff
@@ -312,16 +312,30 @@ static void processing_finished(int err)
 static void edge_impulse_thread_fn(void)
 {
 	signal_t features_signal;
+	int32_t start_time;
 
 	while (true) {
 		k_sem_take(&ei_sem, K_FOREVER);
 
 		features_signal.get_data = &raw_feature_get_data;
 		features_signal.total_length = INPUT_WINDOW_SIZE;
 
+		if (IS_ENABLED(CONFIG_EI_WRAPPER_PROFILING)) {
+			start_time = k_uptime_get_32();
+		}
+
 		/* Invoke the impulse. */
 		EI_IMPULSE_ERROR err = run_classifier(&features_signal,
 						      &ei_result, DEBUG_MODE);
+		if (IS_ENABLED(CONFIG_EI_WRAPPER_PROFILING)) {
+			int32_t delta = k_uptime_get_32() - start_time;
+
+			LOG_INF("run_classifier execution time: %dms", delta);
+			LOG_INF("EI classifier sampling: %dms dsp: %dms classification: %dms",
+				ei_result.timing.sampling,
+				ei_result.timing.dsp,
+				ei_result.timing.classification);
```

Comment:
You could also log timing for anomaly calculation to have complete profiling overview (check if given classifier supports anomaly calculation because it's optional).